storage/prometheus/matrix_selector.go

```diff
@@ -205,7 +207,9 @@ func (o *matrixSelector) Next(ctx context.Context) ([]model.StepVector, error) {
 				} else {
 					vectors[currStep].AppendSample(o.vectorPool, series.signature, f)
 				}
+				stepSamples += int64(len(series.buffer.Samples()))
```

Comment:
Isn't this simply equal to (without the variable):
```
o.IncrementSamplesAtStep(len(series.buffer.Samples()), currStep)
```

Reply:
Thanks for the catch! Fixed it.